ml/pom.xml

```diff
@@ -0,0 +1,346 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright (c) 2021, NVIDIA CORPORATION.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>com.nvidia</groupId>
+        <artifactId>rapids-4-spark-parent</artifactId>
+        <version>21.10.0-SNAPSHOT</version>
+    </parent>
+    <artifactId>rapids-4-spark-ml_2.12</artifactId>
+    <name>RAPIDS Accelerator for Apache Spark ML</name>
+    <description>The RAPIDS cuML library for Apache Spark</description>
+    <version>21.10.0-SNAPSHOT</version>
+
+    <properties>
+        <java.version>1.8</java.version>
+        <spark.version>3.1.2</spark.version>
```

Comment:
why hardcode 3.1.2?

Reply:
We are using some latest (>3.1.0) code from Spark ML side, and test over 3.1.0 and 3.1.2. The code cannot be compiled with versions such as 3.0.1. @rongou  Do we have plan to support earlier version? Please correct me if I miss anything.

Reply:
does it require spark 3.1.2 to run then I assume?  If so it should be documented in readme and really it should only build when building 3.1.2 so the top level pom file can move the module underneath just the 3.1.2 profile.

Reply:
note, I assume its for 3.1.2+ just still means you should pick up the spark version being built but just not build for anything <